scripts/cluster-upgrade.sh

```diff
@@ -73,91 +215,34 @@ then
     fi
 fi
 
-OCP_PATCH="{\"spec\":{\"desiredUpdate\": {\"force\": false, \"image\": \"\", \"version\": \"$OCP_VERSION_TO\"}}}"
-
-TMP_DIR=`mktemp -d`
-if [[ $? -ne 0 ]];
-then
-    echo "ERROR: Couldn't create a temporary directory. Aborting."
-    exit 1
-fi
-
-# kick off the upgrades (or status check)
 for CD_NAMESPACE in `oc get clusterdeployment --all-namespaces | awk '{print $1}' | sort | uniq`;
 do  
     for CD_NAME in `oc -n $CD_NAMESPACE get clusterdeployment -o json | jq -r '.items[] | select(.metadata.labels["api.openshift.com/managed"] == "true") | select(.status.installed == true) | select(.status.clusterVersionStatus.history[0].state == "Completed") | .metadata.name'`;
```

Comment:
This would be so much more efficient if it looped through all the cluster deployments in --all-namespaces and selected on the ones desired, then extracted .metadata.namespace along with .metadata.name, but I cannot for the life of me figure out how to get jq to nicely return the namespace and name in a way that can be used with bash.  ¯\_(ツ)_/¯

Reply:
You could get them both but you'd have to parse the result.

```
oc -n $CD_NAMESPACE get clusterdeployment -o json | jq -r '.items[] | select(.metadata.labels["api.openshift.com/managed"] == "true") | select(.status.installed == true) | select(.status.clusterVersionStatus.history[0].state == "Completed") | .metadata.namespace "," .metadata.name'
```

Maybe there's another way but it's not worth spending time on in my opinion.

Reply:
:+1: